streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -230,6 +230,8 @@ public void onPartitionsRevoked(final Collection<TopicPartition> assignment) {
                 // suspend active tasks
                 suspendTasksAndState();
             } catch (final Throwable t) {
+                // TODO: when this happens, we should still make sure the standby / active tasks list gets updated before returning
```

Comment:
A more general question is that, if we ever got an exception in the `onPartitionRevoked` callback, we will just remember it and proceed as normal, and only re-throw it in the next `poll` call. I.e. we will still execute `onPartitionAssigned` as normal, which smells bad to me.